### PR TITLE
refactor: add slots to references layout

### PR DESCRIPTION
**Pull request by hwkr**
This PR adds a bunch of slots to the API references to allow greater customization.

Specifcally:

- It adds slots for sidebar start and end as well as content start and end
- It removes the footer slot
- It will remove footer below sidebar mode
- It splits `ApiReference.vue` into two components `ApiReference.vue` and `ApiReferenceBase.vue` (open to better names)
  - `ApiReference.vue` maintains the current functionality and API of the references component included search and dark mode toggle
  - `ApiReferenceBase.vue` provides the core functionality of the references and a bunch of slots for customization

Todo:

- [x] ~~Make external dark mode toggling work (@amritk help?)~~ going to tackle in another PR
- [x] Fix mobile view for `ApiReference.vue`

Google Chrome-2023-11-09-15-37-26@2x

### CODE REVIEW

**Comment by hanspagel** [comment]
I’ve had this Collaborative Editing PR draft #431 for a while and had to merge it today. This caused merge conflicts and I tried to resolve them, I was pretty careful and hope I kept all your changes. ✌️

**Comment by hwkr** [comment]
Sounds good! @amritk I'd like to do the hook logic too but I think it should be either another PR or another ticket to keep scope down

**Comment by hwkr** [comment]
Okay so it seems like collaborative editing is working just the dev toolbar is a little buggy @hanspagel 


https://github.com/scalar/scalar/assets/6374090/7e1d7e8b-f537-4727-bd20-cd716814bfed

**Comment by hanspagel** [comment]
That’s good news! Does switching the room work?

**Comment by hwkr** [comment]
> That’s good news! Does switching the room work?

I did not try

File: projects/web/src/pages/ApiReferencePage.vue [hunk 1]
```diff
@@ -1,58 +1,45 @@
...
   spec: {
-    content: window.localStorage.getItem('api-reference-content') ?? undefined,
-    // content: { openapi: '3.1.0', info: { title: 'Example' }, paths: {} },
-    // content: () => {
-    //   return { openapi: '3.1.0', info: { title: 'Example' }, paths: {} }
-    // },
-    // preparsedContent,
-    // url: 'https://raw.githubusercontent.com/outline/openapi/main/spec3.json',
-    // url: 'https://raw.githubusercontent.com/testimio/public-openapi/TES-14404-mobile-applications/api.yaml',
+    url: 'https://raw.githubusercontent.com/outline/openapi/main/spec3.json',
```
Comment (marclave) [inline projects/web/src/pages/ApiReferencePage.vue:16]: should we change this back to 

```
content: window.localStorage.getItem('api-reference-content') ?? undefined,
```

**Comment by marclave** [comment]
wow! this pr is amazing @hwkr , the interface is much cleaner. once we ensure the dev bar is working as expected i can jump in and test all the things like search + toggling ✨

**Comment by hwkr** [comment]
> wow! this pr is amazing @hwkr , the interface is much cleaner. once we ensure the dev bar is working as expected i can jump in and test all the things like search + toggling ✨ 

The dev bar is working as expected! There's two versions now and the standalone version is the one with search and a mobile header.

**Comment by amritk** [comment]
Tested collaborative editing and the syncing seems to work for me
image

File: packages/api-reference/src/components/ApiReference.vue [hunk 2]
```diff
@@ -1,170 +1,66 @@
...
-    props.footerBelowSidebar ||
-    props.isEditable ||
-    props.hocuspocusConfiguration
-  ) {
-    console.warn(
-      '[ApiReference] The <ApiReference /> component now accepts a single `configuration` prop. Please update your code.',
-    )
-  }
+// I don't know why this isn't picking up the v-bind but whatever
+// eslint-disable-next-line vue/no-unused-properties
```
Comment (amritk) [inline packages/api-reference/src/components/ApiReference.vue:13]:
```suggestion
```
I just removed this line with no complaints from lint

File: packages/api-reference/src/components/ApiReference.vue [hunk 3]
```diff
@@ -1,170 +1,66 @@
...
-  emit('updateContent', rawSpecRef.value)
+// Override the sidebar value for mobile to open and close the drawer
+const config = computed(() => {
+  const showSidebar = isMobile.value
+    ? state.showMobileDrawer
+    : props.configuration?.showSidebar
+  const spec: SpecConfiguration = props.configuration?.spec || {
+    content: content.value,
+  }
+  console.log(spec)
```
Comment (amritk) [inline packages/api-reference/src/components/ApiReference.vue:30]:
```suggestion
```

**Review by marclave — APPROVED** [review]
let's go!

**Review by amritk — APPROVED** [review]
Looks good! Nice work!

**Pull request merged**